changelog/dmd.obj_extensions.dd

```diff
@@ -0,0 +1,6 @@
+Accepting .o and .obj file extensions on all platforms
+
+Accepting .o and .obj file extensions on all platforms makes DMD behave
+the same as Clang and other modern compilers. There is no point in
+discarding *.o or *.obj depending on the current OS, as both extensions
+indicate that this is an object file.
```

Comment:
the wording here could be improved. Will come back to this when I am properly awake.